doc/manual/manual.tex

```diff
@@ -5,183 +5,241 @@
+  \ref{sec:shared_library_interface}).}
+  \item \textbf{CTypes interface} (Python): Uses the ALE as an external Python library, 
+  allowing one to write ALE code by just importing the correspondent library. This is a very
+  fast interface (Section \ref{sec:python_interface}).
+  \item \textbf{RL-Glue interface} (C/C++, Java, Python, Matlab or Lisp): Communicates
+  with ALE via RL-Glue (Section \ref{sec:rlglue_interface}).}
+  \item \textbf{FIFO interface} (any language): Communicates with ALE through a text interface 
+  using \verb+stdin/stdout+ (Section \ref{sec:pipes_interface}).}.
+\end{enumerate}
 
-\begin{itemize}
-  \item{\textbf{Shared library interface.} Loads ALE as a shared library (Section \ref{sec:shared_library_interface}).}
-  \item{\textbf{FIFO interface.} Communicates with ALE through a text interface (Section \ref{sec:pipes_interface}).}
-  \item{\textbf{RL-Glue interface.} Communicates with ALE via RL-Glue (Section \ref{sec:rlglue_interface}).}
-\end{itemize}
+We also discuss some sensible features of this environment, such as its stochasticity.
 
-\section{Installing}
+\section{Installing}\label{sec:install}
 
 \subsection{Requirements}
 
-To build and run ALE, you will need:
+The basic requirements to build and run the ALE are:
 
 \begin{itemize}
-  \item{g++ / make}
+  \item \verb+CMake / make / g+++
+\end{itemize}
+Notice that in the past (previous to version 0.5.0) the recommended method to compile the ALE was
+through Makefiles written by the ALE developers. This has changed and now we suggest everyone to
+use CMake to generate a Makefile and then to compile the code using such generated Makefile. The
+old Makefiles written for Linux and OS X (files \verb+makefile.mac+ and \verb+makefile.unix+) are
+still available, but we do not support them anymore.
+
+The ALE provides some functionalities that are not activated by default, but if chosen the user is 
+required to use additional packages:
+\begin{itemize}
+  \item \verb+SDL / RL-Glue+
 \end{itemize}
 
 \subsection{Installation/Compilation}
 
-This tutorial assumes that you have extracted ALE to \verb+ale_0_4+. Compiling
-ALE on a UNIX machine is as simple as:
+One first has to install the ALE requirements before compiling the ALE itself. Assuming that 
+\verb+g+++ and \verb+make+ are already available to the user, to install CMake and SDL is 
+straightforward, and it can be done through package managers in both OS X and Linux:
 
+\subsubsection*{OS X:}
 \begin{verbatim}
-> cd ale_0_4
-ale_0_4> cp makefile.unix makefile
-ale_0_4> make
+  > brew install cmake
+  > brew install sdl
 \end{verbatim}
 
-Alternate makefiles are provided for 
+\subsubsection*{Linux (e.g.: Ubuntu):}
+\begin{verbatim}
+  > sudo apt-get install cmake
+  > sudo apt-get install libsdl1.2-dev
+\end{verbatim}
 
-\begin{itemize}
-  \item{Mac OS X: \verb+makefile.mac+}
-\end{itemize}
+To install RL-Glue, in both systems, we recommend the user to go to the RL-Glue 
```

Comment:
Lets just provide a reference to the RLGlue section which gives more detailed instructions.